File: pharmaSmart-domain/src/main/java/com/kobe/warehouse/domain/enumeration/InventoryCategory.java
package com.kobe.warehouse.domain.enumeration;

public enum InventoryCategory {
    // ── Types de périmètre ────────────────────────────────────────────────────
    /**
     * Tout le magasin (tous produits actifs)
     */
    MAGASIN,
    /**
     * Un storage (point de vente)
     */
    STORAGE,
    /**
     * Un rayon
     */
    RAYON,
    /**
     * Une famille de produits
     */
    FAMILLY,

    // ── Types thématiques (nouveaux) ─────────────────────────────────────────
    /**
     * Produits avec lots périmés et stock > 0 — obligation réglementaire
     */
    PERIME,
    /**
     * Produits dont les lots expirent dans X jours — anticipation retours
     */
    ALERTE_PEREMPTION,
    /**
     * Produits vendus sur une période donnée — vérification cohérence stock/ventes
     */
    VENDU,
    /**
     * Produits sans aucune vente sur une période — stock dormant / déstockage
     */
    INVENDU,
    /**
     * Produits dont le stock est inférieur ou égal au seuil minimum
     */
    SOUS_SEUIL,
    /**
     * Produits en rupture totale (qty_stock = 0)
     */
    EN_RUPTURE,
    /**
     * Selection de produit ids
     */
    SELECTION_PRODUIT,

    GROSSISTE,

    /**
     * Produits de classe Pareto A, B ou C — analyse ABC selon le chiffre d'affaires (vue
     * matérialisée mv_abc_pareto_analysis)
     */
    ABC,

}

File: mobile-inventory/src/main/java/com/kobe/warehouse/inventory/data/model/InventoryCategory.kt
package com.kobe.warehouse.inventory.data.model

import com.google.gson.annotations.SerializedName

/**
 * Inventory Category enum
 * Matches backend com.kobe.warehouse.domain.enumeration.InventoryCategory
 */
enum class InventoryCategory {
    @SerializedName("MAGASIN")
    MAGASIN,

    @SerializedName("STORAGE")
    STORAGE,

    @SerializedName("RAYON")
    RAYON,

    @SerializedName("FAMILLY")
    FAMILLY,

    @SerializedName("PERIME")
    PERIME,

    @SerializedName("ALERTE_PEREMPTION")
    ALERTE_PEREMPTION,

    @SerializedName("VENDU")
    VENDU,

    @SerializedName("INVENDU")
    INVENDU,

    @SerializedName("SOUS_SEUIL")
    SOUS_SEUIL,

    @SerializedName("EN_RUPTURE")
    EN_RUPTURE,

    @SerializedName("SELECTION_PRODUIT")
    SELECTION_PRODUIT,

    @SerializedName("GROSSISTE")
    GROSSISTE,

    @SerializedName("ABC")
    ABC
}

/**
 * Category wrapper as serialized by the backend (CategoryInventory DTO):
 * { "name": "MAGASIN", "label": "Inventaire global" }
 */
data class CategoryInventory(
    @SerializedName("name")
    val name: InventoryCategory? = null,

    @SerializedName("label")
    val label: String? = null
)
